service/raft/raft_group0.cc

```diff
@@ -344,6 +356,12 @@ future<> raft_group0::start_server_for_group0(raft::group_id group0_id) {
     assert(group0_id != raft::group_id{});
 
     auto my_id = co_await load_my_id();
+    auto peers = co_await load_discovered_peers(_qp);
+    for (auto& peer : peers) {
+        if (peer.ip_addr != gms::inet_address{}) {
+            _raft_gr.address_map().add_or_update_entry(peer.id, peer.ip_addr);
```

Comment:
Maybe. But I don't know how to reconcile this. We have now 4 sources of notifications, neither of this can be considered "truth":
- incoming RPCs. They may arrive from an outdated node and update the IP address of this same node running with another IP address. When Raft was used for configuration changes that had been impossible, since changing node's IP address would mean a configuration  change and would require a majority.
- discovery information - as you rightfully noted, it may be outdated
- gossip persistent state - may be outdated as well
- gossip notifications - we may get notifications from the nodes outside Raft configuration. 

We could perhaps introduce "update priorities" or trust levels, and never trust an update from a lower trust source if we got the preceding information from a higher trust source, but then that wouldn't solve the issue of nodes outside Raft configuration gossiping to us its IP addresses. We don't consult Raft configuration when deciding whether or not to trust gossip updates.

Reply:
Added sources of intput to the address map, no override will happen if we have a more fresh input.